src/test/com/fulcrologic/fulcro/algorithms/form_state_spec.cljc

```diff
@@ -148,6 +163,10 @@
   (component "error checking"
     (let [data-tree {:id 1 :name "A" :thing {:id 2 :x 42}}]
       (assertions
+        "throws an exception if any declared form fields are missing from the component query"
+        (fs/add-form-config FormMissingFields data-tree) =throws=> #"to .*FormMissingFields. It declares fields but not all of them are in the query!"
+        "does not throw an exception if all fields are present in the query when some are sub-forms"
```

Comment:
In general I recommend using positive language in test assertions. For example, "Can successfully generate the config for a valid form with subforms".  Saying it doesn't do something opens the door to everything in the universe (conceptually). It doesn't paint my house, wash my car, re-render the screen, launch a rocket, etc...saying what it doesn't do is basically saying nothing at all.  

In the case of exceptions I guess it is a grey area, but I think it is clearer to say that a good case succeeds and a bad case fails.

no need to change anything, I just thought I would pass that on as food for thought.

Reply:
Heh, I had the same thought. If `add-form-config` is used with any subform that would verify this works. I think adding this under a positive test for subforms would be the way to go for something like this in the future.

Thanks for pointing this out!